src/parser/lexer.rs

```diff
@@ -122,19 +122,11 @@ impl<'a, R: CharRead> Lexer<'a, R> {
     }
 
     pub fn eof(&mut self) -> Result<bool, ParserError> {
-        if self.reader.peek_char().is_none() {
```

Comment:
I'm removing this because I think the call to is_not_eof later checks the same (None gets mapped to UnexpectedEof which gets mapped to Ok(true) too).

Tell me if I'm doing something stupid and this needs to stay